databases/week2/Week2-HW.sql

```diff
@@ -0,0 +1,358 @@
+values (1, 1),
+    (1, 2),
+    (2, 3),
+    (2, 4),
+    (3, 5),
+    (3, 2);
+
+-- Homework tasks for week2
+-- Part 1: Working with tasks
+--  a) Add a task with these attributes: title, description, created, updated, due_date, status_id, user_id
+insert into
+    task (
+        title,
+        description,
+        created,
+        updated,
+        due_date,
+        status_id,
+        user_id
+    )
+values (
+        'Studying',
+        'Complete the week3 database preparation',
+        '2024-07-08 00:23:58',
+        '2024-07-10 12:00:25',
+        '2025-07-08 00:00:23',
+        1,
+        1
+    );
+-- b) Change the title of a task
+update task set title = 'Learning' ORDER BY updated DESC limit 1;
+-- c) Change a task due date
+update task
+set
+    due_date = '2024-12-12 00:02:23'
+where
+    id = 36;
+-- d) Change a task status
+update task set status_id = 2 ORDER BY updated desc limit 1;
+-- e) Mark a task as complete
+update task set status_id = 3 ORDER BY updated desc limit 1;
+-- f) Delete a task
+delete from task ORDER BY updated desc limit 1;
+
+/* Part 2: School database
+Create a new database containing the following tables:
+Class: with the columns: id, name, begins (date), ends (date)
+Student: with the columns: id, name, email, phone, class_id (foreign key) */
+
+create database school_1;
+
+create table class (
+    id int PRIMARY key AUTO_INCREMENT,
+    name varchar(50),
+    begin_date DATE,
+    end_date DATE
+);
+
+create table student_1 (
+    id int PRIMARY key AUTO_INCREMENT,
+    name varchar(50),
+    email varchar(100),
+    phone int(10),
+    class_id int,
+    constraint class_id_fk Foreign Key (class_id) REFERENCES class (id)
+);
+/* If you are done with the above tasks, you can continue with these advanced tasks:
+Create an index on the name column of the student table.
+Add a new column to the class table named status which can only have the following values: not-started, ongoing, finished (hint: enumerations) */
+create index name_index on student_1 (name);
+
+alter table class
+add COLUMN status ENUM(
+    'not-started',
+    'ongoing',
+    'finished'
+);
+-- Inserting values into the tables class and student_1
+insert into
+    class
+values (
+        1,
+        'John',
+        '2024-01-01',
+        '2025-01-01',
+        'ongoing'
+    ),
+    (
+        2,
+        'James',
+        '2023-01-01',
+        '2024-01-01',
+        'finished'
+    );
+
+update class set name = 'Maths' where id = 1;
+
+update class set name = 'physics' where id = 2;
+
+insert into
+    student_1
+values (
+        1,
+        'John',
+        'john_01@gmail.com',
+        '246810235',
+        1
+    );
+
+insert into
+    student_1
+values (
+        2,
+        'James',
+        'James_01@gmail.com',
+        '2536985',
+        2
+    );
+
+insert into
+    student_1
+values (
+        3,
+        'Smith',
+        'smith_01@gmail.com',
+        '136856985',
+        2
+    );
+-- Part 3: More queries
```

Comment:
The queries from this part seem fine BUT you used other tables from the ones specified on the assignment.

Reply:
Yes, Sorry for my overlook. I have updated the database (user_task table) and changed the queries based on that